gammapy/data/tests/test_data_store.py

```diff
@@ -257,6 +257,16 @@ def test_data_store_bad_name():
         DataStore.from_dir("$GAMMAPY_DATA/hess-dl3-dr1/", "hdu-index.fits.gz", "bad")
 
 
+@requires_data()
+def test_data_store_doubled_id():
+    """Test that doubled id in obs_id causes UserWarning"""
+
+    # Using example provided in Issue #5943:
+    datastore = DataStore.from_dir("$GAMMAPY_DATA/hess-dl3-dr1")
+    with pytest.raises(UserWarning):
+        datastore.get_observations(obs_id=[23523, 23523])
```

Comment:
It can be useful to check the exact warning message, see https://docs.gammapy.org/dev/development/dev_howto.html#caplog-fixture